folium/features.py

```diff
@@ -288,21 +282,63 @@ def render(self, **kwargs):
             </style>
             """).render(this=self, **kwargs)), name=self.get_name())
 
-        figure.header.add_child(
-            JavascriptLink('https://d3js.org/d3.v3.min.js'),
-            name='d3')
+        if vegalite_major_version == '1':
+            self._embed_vegalite_v1(figure)
+        elif vegalite_major_version == '2':
+            self._embed_vegalite_v2(figure)
+        elif vegalite_major_version == '3':
+            self._embed_vegalite_v3(figure)
+        else:
+            # Version 2 is assumed as the default, if no version is given in the schema.
+            self._embed_vegalite_v2(figure)
 
-        figure.header.add_child(
-            JavascriptLink('https://cdnjs.cloudflare.com/ajax/libs/vega/2.6.5/vega.min.js'),  # noqa
-            name='vega')
+    def _get_vegalite_major_versions(self, spec):
+        try:
+            schema = spec['$schema']
+            version = os.path.splitext(os.path.split(schema)[1])[0].lstrip('v')
```

Comment:
Do you need `os` to split `schema` here? I'm just thinking that we can avoid one extra import.

Reply:
I think one could solve it just with `str.split`, but the `os` functions more direct and have somewhat speaking names. I also think importing `os` does not harm as it is in the standard library.

Reply:
I'm +0.5 on that. Indeed `os` is standard lib, but it is an extra import that is used for a simple task. I would also argue that .split('/') is more readble than `.path.splitext` which is intended for file system paths.

Reply:
Convinced! Even needs one method call less this way.